js/CompanyInfo.js

```diff
@@ -0,0 +1,137 @@
+            </div>`;
+        
+        const canvas = document.getElementById('companyChart')
+        const ctx = canvas.getContext("2d");
+        ctx.textAlign = "center";
+        ctx.fillStyle = 'rgb(255, 99, 132)';
+        ctx.fillText("Loading graph...", canvas.width/2, canvas.height/2);
+
+    }
+
+    renderCompanyData(data) {
+        let redClassIfNegative = '';
+        const changeInPercentFormated = parseFloat(data.changesPercentage).toFixed(2)
+        const priceTwoDecimals = parseFloat(data.price).toFixed(2)
+        if (changeInPercentFormated < 0) { redClassIfNegative = 'red'; }
+        this.dataContainer.innerHTML = `
+          <div class="headline mb-4 text-center d-flex align-items-center">
+            <a id=comp-website href="${data.website}" target="blank" class="d-flex align-items-center">
+              <img id="comp-logo" class="headline-logo" src="${data.image}" alt="">
+              <h2 class="mb-0 text-center display-6">${data.companyName}</h2>
+            </a>
+          </div>
+          <h3 class="stock-price mb-3 d-flex align-items-center">
+            Stock price:&nbsp;
+            <span id="comp-price">${priceTwoDecimals}</span>&nbsp;
+            <span id="comp-currency">${data.currency}</span> &nbsp;
+            <span class="comp-change ${redClassIfNegative}" id="comp-change">${changeInPercentFormated}</span>
+          </h3>
+          <p id="comp-desp">${data.description}</p>
+        `;
+    }
+
+    async addChart() {
+        const url = CompanyInfo.apiChartUrl + this.symbol + CompanyInfo.apiChartQueryString;
+        const historicDataArray = await this.makeApiRequest(url);
+        const chartData = this.getChartData(historicDataArray);
+        this.renderChart(chartData);
+    }
+
+    
+    renderChart(data=[]) {
+
+        const config = {
+            type: 'line',
+            data: {
+                //labels: dates,
+                datasets: [{
+                    label: 'Stock Price History',
+                    fill: 'origin',
+                    backgroundColor: 'rgb(255, 99, 132)',
+                    borderColor: 'rgb(255, 99, 132)',
+                    pointRadius: 0,
+                    pointHoverRadius: 15,
+                    //data: prices
+                    data: data
+                }]
+            },
+            options: {}
+        };
+
+        const myChart = new Chart(
+            document.getElementById('companyChart'),
+            config
+        );
+
+    }
+
+    getChartData(historicDataArray) {
+        let data = historicDataArray.historical.map((item) => {
+            return { 'x': item.date, 'y': item.close };
+        });
+        if (data.length > 1 && data.length > CompanyInfo.graphPointMaxQty) {
+            const filterDivNum = Math.ceil(data.length / CompanyInfo.graphPointMaxQty);
+            data = data.filter((element, index) => index % filterDivNum == 0);
+        }
+        return data;
+    }
+
+
+    async makeApiRequest(url) {
```

Comment:
this funciton is alredy written in another file. you should create a common file for functions and varibles that are being used in more than 1 class/file.